### ISSUE TITLE
Can't pass gradient through jax.random.bernoulli

### ISSUE THREAD

**Issue by gcgibson** [google/jax#11606]
Hi, 

I've been trying to build a simple differentiable agent-based model in JAX, but I can't get the gradient to be non-zero when passed through a random call (specifically Bernoulli).

```import random as random
import jax.numpy as jnp
import jax
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import networkx as nx
from jax import random
key = random.PRNGKey(0)

def abm_native(n_agents = 100, expected_degree_sequence = None, params=.5, gamma=.2):
    
    new_inf = jax.random.bernoulli(key,p=jnp.repeat(params,n_agents))
    return (jnp.sum(1.0*new_inf))
    


x = abm_native(n_agents=100,expected_degree_sequence=None)
print (x)

def forward(params):
    x = abm_native(n_agents=100,expected_degree_sequence=None,params=params)
    return (x)





def loss_fn(params, x):
    err = forward(params) - x
    return jnp.mean(jnp.square(err))  # m  

from jax import grad, jit, vmap

grad_fn = jax.grad(loss_fn)


def update(params, grads):
    return jax.tree_map(lambda p, g: p - 0.005 * g, params, grads)

params = .3
# the main training loop
for _ in range(100):
    loss = loss_fn(params, x)
    print("loss",loss)

    grads = grad_fn(params,x)
    params = update(params, grads)
    print ("params",params)
    print ("grads",grads)```
